### PR TITLE
return_distances kwarg to self_capped_distance 

### LINKED ISSUE #2101: Add return_distances kwarg to self_capped_distance
The function `lib.distances.self_capped_distance` offers a quick way to evaluate pairwise distances within a set of coordinates.  This returns indices and distances of pairs of coordinates which are within a certain threshold.

Its partner function `lib.distances.capped_distance` does similar but between two arrays of coordinates.  This function has an optimisation that allows the distances to not be returned (kwarg `return_distances=False`)

TODO:
 - [ ] write tests
 - [ ] add `return_distances=False` to `self_capped_distance`
 - [ ] verify that this is faster

### PR DESCRIPTION
**Pull request by NinadBhat**
Fixes #2101 
Changes made in this Pull Request:
 - Added return_distances kwarg to self_capped_distance 

I am not sure how I  should verify that this change makes self_capped_distance is faster. Kindly let me know how to do that.

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/lib/distances.py [hunk 1]
```diff
@@ -1064,12 +1086,16 @@ def _nsgrid_capped_self(reference, max_cutoff, min_cutoff=None, box=None):
             results = gridsearch.self_search()
 
         pairs = results.get_pairs()[::2, :]
-        distances = results.get_pair_distances()[::2]
-        if min_cutoff is not None:
-            idx = distances > min_cutoff
-            pairs, distances = pairs[idx], distances[idx]
+        if return_distances or (min_cutoff is not None):
```
Comment (zemanj) [inline package/MDAnalysis/lib/distances.py]: What's the reason for having `or (min_cutoff is not None)` here?

File: package/MDAnalysis/lib/distances.py [hunk 2]
```diff
@@ -1064,12 +1086,16 @@ def _nsgrid_capped_self(reference, max_cutoff, min_cutoff=None, box=None):
...
+            distances = results.get_pair_distances()[::2]
+            if min_cutoff is not None:
+                idx = distances > min_cutoff
+                pairs, distances = pairs[idx], distances[idx]
 
-    return pairs, distances
+    if return_distances:
+        return pairs, distances
+    else:
```
Comment (zemanj) [inline package/MDAnalysis/lib/distances.py]: You can omit the `else` here and dedent the following line instead

File: package/MDAnalysis/lib/distances.py (hunk 1 above)
Comment (zemanj) [reply package/MDAnalysis/lib/distances.py]: Sorry I got it now. :+1:

**Comment by zemanj** [comment]
I don't quite understand why CodeCov marks line 997 and 1089 (`if return_distances or (min_cutoff is not None):`) as partial hits since you added tests for all possible combinations with `return_distances` set to `False` or `True`... any ideas?

**Comment by zemanj** [comment]
> I am not sure how I should verify that this change makes self_capped_distance is faster. Kindly let me know how to do that.

You could just run `self_capped_distance` locally with and without `return_distances` using positions from an arbitrary test universe (you can use one from `MDAnalysisTests.datafiles` if you don't have any simulation data at hand).
Or, even nicer, you could add corresponding `asv` benchmarks in `<repo_root>/benchmarks/benchmarks/analysis/distances.py`!

**Comment by richardjgowers** [comment]
@NinadBhat also add yourself to AUTHORS and add a CHANGELOG entry please

File: testsuite/MDAnalysisTests/lib/test_distances.py [hunk 3]
```diff
@@ -171,14 +171,20 @@ def test_capped_distance_return(npoints, box, query, method, min_cutoff):
 @pytest.mark.parametrize('box', boxes_1)
 @pytest.mark.parametrize('method', method_1)
 @pytest.mark.parametrize('min_cutoff', min_cutoff_1)
-def test_self_capped_distance(npoints, box, method, min_cutoff):
+@pytest.mark.parametrize('ret_dist', (False, True))
+def test_self_capped_distance(npoints, box, method, min_cutoff, ret_dist):
```
Comment (zemanj) [inline testsuite/MDAnalysisTests/lib/test_distances.py]: @NinadBhat I tracked down the reason for the partial hits for you. Hopefully CodeCov is happy now... :crossed_fingers:

Comment (zemanj) [reply testsuite/MDAnalysisTests/lib/test_distances.py]: 100% diff coverage. Woohooo! :tada:

**Comment by richardjgowers** [comment]
@NinadBhat awesome first contribution, thanks!

**Comment by zemanj** [comment]
>awesome first contribution, thanks!

I second that! :+1:

**Pull request merged**